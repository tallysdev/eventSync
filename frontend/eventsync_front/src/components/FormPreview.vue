<template>
    <v-form class="d-flex flex-column align-center">
        <div v-for="(question, index) in questions" :key="index" class="field-size">
            <v-card outlined elevation="4">
            <v-card-title class="d-flex justify-space-between align-center">
                <p class="question-text">{{ question.text }}</p>
            </v-card-title>
    
            <v-card-subtitle v-if="question.type === 'Discursiva'">
                <v-textarea readonly :placeholder="`Resposta para: ${question.text}`" outlined></v-textarea>
            </v-card-subtitle>
    
            <v-card-subtitle v-if="question.type === 'Múltipla escolha'">
                <v-checkbox v-for="(option, optIndex) in question.optionList" :key="optIndex" :label="option" :value="option"></v-checkbox>
            </v-card-subtitle>
    
            <v-card-subtitle v-if="question.type === 'Objetiva'">
                <v-radio-group v-model="question.selectedOption" column>
                    <v-radio v-for="(option, optIndex) in question.optionList" :key="optIndex" :label="option" :value="option"></v-radio>
                </v-radio-group>
                </v-card-subtitle>
            </v-card>
        </div>
    </v-form>
</template>
  
<script setup lang="ts"> 
import type { QuestionView } from '@/types/questions';

const props = defineProps<{
    questions: QuestionView[]
}>()
</script>
  
<style scoped>
.field-size {
    max-width: 350px;
    min-width: 300px;
    width: 100%;
    margin-bottom: auto;
    font-size: 1.2rem;

}

.question-text {
    word-wrap: break-word; /* Garante que o texto longo seja quebrado */
    overflow-wrap: break-word; /* Garante que o texto longo seja quebrado */
    white-space: pre-wrap; /* Mantém espaços e quebras de linha */
}
</style>
  