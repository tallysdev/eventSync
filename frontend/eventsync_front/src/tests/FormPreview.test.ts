import { mount } from '@vue/test-utils'
import { describe, it, expect } from 'vitest'
import FormPreview from '@/components/FormPreview.vue'

interface QuestionView {
  text: string
  type: 'Discursiva' | 'Múltipla escolha' | 'Objetiva'
  optionList: string[]
  selectedOptions: string[]
  selectedOption: string
}

describe('FormPreview.vue', () => {
  const questions: QuestionView[] = [
    {
      text: 'O que você mais gostou?',
      type: 'Discursiva',
      optionList: [],
      selectedOptions: [],
      selectedOption: ''
    },
    {
      text: 'Escolha seu tipo de evento:',
      type: 'Múltipla escolha',
      optionList: ['AAA', 'BBB', 'CCC'],
      selectedOptions: [],
      selectedOption: ''
    },
    {
      text: 'Qual é o seu gênero musical favorito?',
      type: 'Objetiva',
      optionList: ['Rock', 'Pop', 'Jazz'],
      selectedOptions: [],
      selectedOption: ''
    }
  ]

  const wrapper = mount(FormPreview, {
    props: {
      questions
    }
  })

  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders the correct number of questions', () => {
    const questionCards = wrapper.findAll('v-card')
    expect(questionCards.length).toBe(questions.length)
  })
})
