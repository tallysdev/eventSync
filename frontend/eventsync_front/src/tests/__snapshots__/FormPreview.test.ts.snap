// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FormPreview.vue > renders correctly 1`] = `
"<v-form data-v-7395e4fa="" class="d-flex flex-column align-center">
  <div data-v-7395e4fa="" class="field-size">
    <v-card data-v-7395e4fa="" outlined="" elevation="4">
      <v-card-title data-v-7395e4fa="" class="d-flex justify-space-between align-center">
        <p data-v-7395e4fa="" class="question-text">O que você mais gostou?</p>
      </v-card-title>
      <v-card-subtitle data-v-7395e4fa="">
        <v-textarea data-v-7395e4fa="" readonly="" placeholder="Resposta para: O que você mais gostou?" outlined=""></v-textarea>
      </v-card-subtitle>
      <!--v-if-->
      <!--v-if-->
    </v-card>
  </div>
  <div data-v-7395e4fa="" class="field-size">
    <v-card data-v-7395e4fa="" outlined="" elevation="4">
      <v-card-title data-v-7395e4fa="" class="d-flex justify-space-between align-center">
        <p data-v-7395e4fa="" class="question-text">Escolha seu tipo de evento:</p>
      </v-card-title>
      <!--v-if-->
      <v-card-subtitle data-v-7395e4fa="">
        <v-checkbox data-v-7395e4fa="" label="AAA" value="AAA"></v-checkbox>
        <v-checkbox data-v-7395e4fa="" label="BBB" value="BBB"></v-checkbox>
        <v-checkbox data-v-7395e4fa="" label="CCC" value="CCC"></v-checkbox>
      </v-card-subtitle>
      <!--v-if-->
    </v-card>
  </div>
  <div data-v-7395e4fa="" class="field-size">
    <v-card data-v-7395e4fa="" outlined="" elevation="4">
      <v-card-title data-v-7395e4fa="" class="d-flex justify-space-between align-center">
        <p data-v-7395e4fa="" class="question-text">Qual é o seu gênero musical favorito?</p>
      </v-card-title>
      <!--v-if-->
      <!--v-if-->
      <v-card-subtitle data-v-7395e4fa="">
        <v-radio-group data-v-7395e4fa="" modelvalue="" column="">
          <v-radio data-v-7395e4fa="" label="Rock" value="Rock"></v-radio>
          <v-radio data-v-7395e4fa="" label="Pop" value="Pop"></v-radio>
          <v-radio data-v-7395e4fa="" label="Jazz" value="Jazz"></v-radio>
        </v-radio-group>
      </v-card-subtitle>
    </v-card>
  </div>
</v-form>"
`;
